### PR TITLE
[msal-node] Fix the device code cancellation

### PR DESCRIPTION
**Pull request by samuelkubai**
Use `Object.assign()` instead of object de-structuring to ensure the request object remains linked to the original request object. To allow for the cancellation flag.

### CODE REVIEW

File: samples/msal-node-samples/standalone-samples/device-code/index.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ const deviceCodeRequest = {
     deviceCodeCallback: (response) => (console.log(response.message)),
     scopes: ["user.read"],
     timeout: 20,
+    cancel: false,
```
Comment (sameerag) [inline samples/msal-node-samples/standalone-samples/device-code/index.js:21]: So how do we make cancel=true? What is the usage? When does an app know it should not poll anymore?

Comment (samuelkubai) [reply samples/msal-node-samples/standalone-samples/device-code/index.js:21]: Cancel is designed to be set `true` after executing this request, on this sample you can find that scenario [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/74f2dfce8804e25035cd3ba21d195530a3fba450/samples/msal-node-samples/standalone-samples/device-code/index.js#L30) where we set a timeout to set the `cancel` to `true` after a few seconds.

Comment (samuelkubai) [reply samples/msal-node-samples/standalone-samples/device-code/index.js:21]: The usage is when the request is made and we are polling, at any point after polling has begun the user of the library can program a cancel by changing the flag from `false` to `true`

**Comment by jasonnutter** [comment]
Fix looks good, is it possible to add a test that demonstrates the bug and fix?

**Comment by samuelkubai** [comment]
Thanks for pointing it out @jasonnutter, I'll work on adding a test for this fix.

**Comment by samuelkubai** [comment]
@jasonnutter, I'm having a few issues adding the necessary e2e test and thus propose to merge this pull request to unblock the dependents while I work on adding the test scenario.

**Comment by jasonnutter** [comment]
> @jasonnutter, I'm having a few issues adding the necessary e2e test and thus propose to merge this pull request to unblock the dependents while I work on adding the test scenario.

I see, no problem. Any unit tests you can add?

**Comment by jasonnutter** [comment]
> > > @jasonnutter, I'm having a few issues adding the necessary e2e test and thus propose to merge this pull request to unblock the dependents while I work on adding the test scenario.
> > 
> > 
> > I see, no problem. Any unit tests you can add?
> 
> @jasonnutter I think this use case is best covered with a e2e test

Makes sense.

**Review by jasonnutter — APPROVED** [review]

**Review by sameerag — APPROVED** [review]
lgtm

**Comment by ramya-rao-a** [comment]
Hello!
Do we know when this fix will be released?

cc @sadasant

**Comment by sameerag** [comment]
@ramya-rao-a We plan to make a release this week, waiting on one more PR. Mostly in a day or two.

**Pull request merged**